bundles/org.pitest.pitclipse.launch.ui/src/org/pitest/pitclipse/launch/ui/PitLaunchShortcut.java

```diff
@@ -135,7 +135,14 @@ private void showNoTestsFoundDialog() {
     private void performLaunch(IJavaElement element, String mode) throws InterruptedException, CoreException {
         ILaunchConfigurationWorkingCopy tmp = createLaunchConfiguration(element);
         Optional<ILaunchConfiguration> existingConfig = findExistingLaunchConfiguration(tmp);
-        ILaunchConfiguration config = existingConfig.orElse(tmp.doSave());
+        ILaunchConfiguration config = existingConfig.orElseGet(() -> {
+            try {
+                return tmp.doSave();
+            } catch (CoreException e) {
```

Comment:
Unfortunately, Eclipse API is not "lambda ready" due to checked exceptions like CoreException, and that case is not covered (as new code) https://sonarcloud.io/component_measures?id=org.pitest%3Aorg.pitest.pitclipse&metric=new_coverage&pullRequest=185&selected=org.pitest%3Aorg.pitest.pitclipse%3Abundles%2Forg.pitest.pitclipse.launch.ui%2Fsrc%2Forg%2Fpitest%2Fpitclipse%2Flaunch%2Fui%2FPitLaunchShortcut.java&view=list

Besides not covered I think this solution is bad (that is, returning null) because

- it will generate an NPE in that case (as you write in the comment)
- we don't benefit from the `throws` declaration that already takes care of propagating such an exception

For these reasons, I'd suggest using the "old school" style, instead of a lambda, e.g., something like `if (config.isPresent()).